code/go/0chain.net/smartcontract/storagesc/handler.go

```diff
@@ -757,78 +757,48 @@ func (ssc *StorageSmartContract) GetValidatorHandler(ctx context.Context,
 
 }
 
-type BlobberOpenChallengesResponse struct {
-	BlobberID                string            `json:"blobber_id"`
-	ChallengeIDs             []string          `json:"challenge_ids"`
-	LatestCompletedChallenge *StorageChallenge `json:"lastest_completed_challenge"` // TODO: fix typo with Blobber and gosdk
+type StorageChallengeInfo struct {
+	ID             string            `json:"id"`
+	Created        common.Timestamp  `json:"created"`
+	Validators     []*ValidationNode `json:"validators"`
+	RandomNumber   int64             `json:"seed"`
+	AllocationID   string            `json:"allocation_id"`
+	AllocationRoot string            `json:"allocation_root"`
+	BlobberID      string            `json:"blobber_id"`
+	Responded      bool              `json:"responded"`
+}
+
+type ChallengesInfo struct {
+	BlobberID  string                  `json:"blobber_id"`
+	Challenges []*StorageChallengeInfo `json:"challenges"`
 }
 
 func (ssc *StorageSmartContract) OpenChallengeHandler(ctx context.Context, params url.Values, balances cstate.StateContextI) (interface{}, error) {
 	blobberID := params.Get("blobber")
 
-	// return "404", if blobber not registered
-	blobber := StorageNode{ID: blobberID}
-	if err := balances.GetTrieNode(blobber.GetKey(ssc.ID), &blobber); err != nil {
+	blobber, err := balances.GetEventDB().GetBlobber(blobberID)
+	if err != nil {
 		return "", smartcontract.NewErrNoResourceOrErrInternal(err, true, "can't find blobber")
 	}
 
-	rsp := &BlobberOpenChallengesResponse{
-		BlobberID:    blobberID,
-		ChallengeIDs: []string{},
-	}
-
-	// return "200" with empty list, if no challenges are found
-	blobberChallenges := &BlobberChallenges{
-		BlobberID: blobberID,
-	}
-	err := blobberChallenges.load(balances, ssc.ID)
-	switch err {
-	case util.ErrValueNotPresent:
-		return rsp, nil
-	case nil:
-		lfb := balances.GetLatestFinalizedBlock()
-		if lfb == nil {
-			return nil, common.NewErrInternal("chain is not ready, could not get latest finalized block")
-		}
-
-		cct := getMaxChallengeCompletionTime()
-		ocs := blobberChallenges.GetOpenChallengesNoExpire(lfb.CreationDate, cct)
-		if len(ocs) > 0 {
-			rsp.ChallengeIDs = make([]string, len(ocs))
-			for i, oc := range ocs {
-				rsp.ChallengeIDs[i] = oc.ID
-			}
-		}
-
-		return rsp, nil
-	default:
-		return nil, common.NewErrInternal("fail to get blobber challenge", err.Error())
+	challenges, err := getOpenChallengesForBlobber(blobberID, common.Timestamp(blobber.ChallengeCompletionTime), balances)
```

Comment:
Get challenge ids of the blobber from the index of blobberChallenges first, then get the blobbers from event db without applying conditions.